application/models/Traits/PermissionTrait.php

```diff
@@ -3,50 +3,109 @@
 trait PermissionTrait
 {
     /**
-     * Get the owner id of this record
+     * @inheritdoc
      * Used for Permission, to be extendable for each model with owner
-     * @return integer|null
      */
     public function getOwnerId()
     {
         return null;
     }
 
     /**
-     * Get Permission data for Permission object
-     * @param string $key
-     * @return array
+     * @inheritdoc
      */
     public static function getPermissionData()
     {
         return array();
     }
 
     /**
-     * Get minimal permission name (for read value)
-     * @return null|string
+     * @inheritdoc
      */
     public static function getMinimalPermissionRead()
     {
         return null;
     }
 
     /**
-     * Get the permission of current model
-     * @param string $sPermission Name of the permission
-     * @param string $sCRUD The permission detail you want to check on: 'create','read','update','delete','import' or 'export'
-     * @param integer $iUserID User ID - if not given the one of the current user is used
-     * @return boolean
+     * @inheritdoc
+     * Criteria used for list or grid
+     * Avoid usage of Permission on each object returned when there are a lot.
+     * @see self::withListRight for adding it at scope in find action.
+     * @param integer $userid (use current if is null)
+     * @return CDbCriteria
+     */
+    public static function getPermissionCriteria($userid = null)
```

Comment:
Not sure if this should be done now or not. Don't recall if already discussed or not. If so, please disregard.

But shouldn't this be called `getListPermissionCritria()` or something that contains the word `List`?
That would inform what kind of Permission criteria is returned.

Reply:
In Survey : we already have scope[ `permission` before](https://github.com/SondagesPro/LimeSurvey-SondagesPro/blob/3394d3d84ae811567e33e2fdafb2c02002290dae/application/models/Survey.php#L261) … i deprecae if 

This function already ezist in Survey and SurveysGroup … maybe it's not the good time to rename it ? 

I can … but i really think we goes far far away from the original issue now …

Reply:
Maybe just leave a todo?

Reply:
> Maybe just leave a todo?

Unsure : it's really near Permission, since if this minimum part are not done : we must send disallow all other access.

@olleharstedt any position ? 

I can deprecate `getPermissionCriteria` to `getListPermissionCriteria` but : without List right : no permission .

Reply:
Something for our biweekly discussion meetings that haven't happened yet...?

Reply:
There are a biweekly discussion meetings now ?

Last time i come, i quit after 10 minutes

Reply:
There should be, with me, you, Gabriel, Carsten.

Reply:
Where . On cliq or discord ?
And when ?

Reply:
Dunno :( Tool doesn't matter much, but we need time.

Reply:
ping

Reply:
Reading this for a ticket... is this method expected to be overriden? If this is a trait can that be done?

Reply:
> Reading this for a ticket... is this method expected to be overriden? If this is a trait can that be done?

In general yes, minimal by ownerid

But : clearly i'm tired here … it's a **more than one year started commit** … 

I really think we NEED a cleaner solution to get permission criteria … but maybe i'm the only one to care of such solution …

It's already the second (or third unsure) time i see we need a scope for permission on listing … 
And after : we can add clear event to update Listing by plugin (needed currently)

Reply:
> > Reading this for a ticket... is this method expected to be overriden? If this is a trait can that be done?
> 
> In general yes, minimal by ownerid


I think methods on traits are not to be overriden.
So overriding this method (`getPermissionCriteria`) will not be straightforward.